gordon/main.py

```diff
@@ -83,20 +86,76 @@ def setup(config_root=''):
     return config
 
 
-def _log_or_exit_on_exceptions(errors, debug):
+def _log_or_exit_on_exceptions(base_msg, exc, debug):
     log_level_func = logging.warn
     if not debug:
         log_level_func = logging.error
 
-    base_msg = 'Plugin "{name}" was not loaded:'
-    for name, exc in errors:
-        msg = base_msg.format(name=name)
-        log_level_func(msg, exc_info=exc)
+    if isinstance(exc, list):
+        for exception in exc:
+            log_level_func(base_msg, exc_info=exception)
+    else:
+        log_level_func(base_msg, exc_info=exc)
 
     if not debug:
         raise SystemExit(1)
 
 
+def _assert_required_plugins(plugins, debug):
+    required_providers = {
+        'event_consumer': False,
+        'enricher': False,
+        'publisher': False,
+    }
+    for plugin in plugins:
+        if interfaces.IEventConsumerClient.providedBy(plugin):
+            required_providers['event_consumer'] = True
+        elif interfaces.IEnricherClient.providedBy(plugin):
+            required_providers['enricher'] = True
+        elif interfaces.IPublisherClient.providedBy(plugin):
+            required_providers['publisher'] = True
+
+    missing = []
+    msg = ('The provider for "{name}" interface is not configured for the '
```

Comment:
for "{name}" -> for the "{name}"

Reply:
bump

Reply:
damnit I swore I got this one. maybe didn't commit it.